lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart

```diff
@@ -2225,28 +2256,95 @@ class MailboxDashBoardController extends ReloadableController {
     setSelectedEmail(email.toPresentationEmail());
   }
 
+  void _handleRestoreDeletedMessageSuccess(EmailRecoveryActionId? emailRecoveryActionId) async {
+    log('MailboxDashBoardController::_handleRestoreDeletedMessageSuccess():emailRecoveryActionId: $emailRecoveryActionId');
+    consumeState(
+      _getMailboxByRoleInteractor.execute(
+        sessionCurrent!,
+        accountId.value!,
+        Role('Restored-Messages')
+      )
+    );
```

Comment:
I still not understand the concept of this interactor, please move it out of this logic

Reply:
dont combine `consumeState` in controller

Reply:
move the logic to get `recovered` folder to the get status of recovery task (only when complete)